drivers/spi/spi_sam.c

```diff
@@ -467,11 +699,22 @@ static const struct spi_driver_api spi_sam_driver_api = {
 	.release = spi_sam_release,
 };
 
-#define SPI_SAM_DEFINE_CONFIG(n)					\
-	static const struct spi_sam_config spi_sam_config_##n = {	\
-		.regs = (Spi *)DT_INST_REG_ADDR(n),			\
-		.periph_id = DT_INST_PROP(n, peripheral_id),		\
-		.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(n),		\
+#define SPI_DMA_INIT(n)                                                                            \
+	.dma_dev = DEVICE_DT_GET(DT_INST_DMAS_CTLR_BY_NAME(n, tx)),                                \
+	.dma_tx_channel = DT_INST_DMAS_CELL_BY_NAME(n, tx, channel),                               \
+	.dma_tx_perid = DT_INST_DMAS_CELL_BY_NAME(n, tx, perid),                                   \
+	.dma_rx_channel = DT_INST_DMAS_CELL_BY_NAME(n, rx, channel),                               \
+	.dma_rx_perid = DT_INST_DMAS_CELL_BY_NAME(n, rx, perid),
+
+#define SPI_SAM_USE_DMA(inst) DT_INST_DMAS_HAS_NAME(n, tx) && IS_ENABLED(CONFIG_SPI_SAM_DMA)
+
+#define SPI_SAM_DEFINE_CONFIG(n)                                                                   \
+	static const struct spi_sam_config spi_sam_config_##n = {                                  \
+		.regs = (Spi *)DT_INST_REG_ADDR(n),                                                \
+		.periph_id = DT_INST_PROP(n, peripheral_id),                                       \
+		.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(n),                                         \
+		.loopback = DT_INST_PROP_OR(n, loopback, 0),                                       \
```

Comment:
A boolean property can be used as is.

```suggestion
		.loopback = DT_INST_PROP(n, loopback),                                             \
```

Reply:
I was out of date regarding booleans. We can go with `DT_PROP`.